core/opl/shovel.py

```diff
@@ -83,18 +120,178 @@ def args(parser, group_actions):
             title="opensearch",
             description="Options needed to work with OpenSearch",
         )
+        group.add_argument(
+            "--es-index",
+            default="rhtap-ci-status-data",
+            help="Elastic search index where the data will be stored",
+        )
+        group.add_argument("--data-file", help="json file to upload to elastic search")
+        group.add_argument("--end-timestamp", help="timestamp when the test ended")
 
 
 class pluginHorreum:
     def __init__(self, args):
         print("Hello from pluginHorreum init")
         self.logger = logging.getLogger("opl.showel.pluginHorreum")
 
-    def upload(self):
-        print("Hello from pluginHorreum upload")
+    def upload(self, args):
+        if args.horreum_data_file is None:
+            raise Exception(
+                "Horreum data file is required to work with --horreum-upload"
+            )
+        elif args.horreum_host is None:
+            raise Exception("Horreum host is required to work with --horreum-upload")
+        elif args.token is None:
+            raise Exception(
+                "Authorisation Token is required to work with --horreum-upload"
+            )
+        elif args.job_name is None:
+            raise Exception("Job name is required to work with --horreum-upload")
+        else:
+            headers = {
+                "Content-Type": "application/json",
+                "Authorization": f"Bearer {args.token}",
+            }
+            jsonFile = open(args.horreum_data_file, "r")
+            values = json.load(jsonFile)
+            jsonFile.close()
+            test_matcher = values[args.test_job_matcher]
+            test_start = values["timestamp"]
```

Comment:
Let's not rely on start/end time existing in the JSON, make user to provide it on command line. Same for `$schema`.